src/platform/macos/view.rs

```diff
@@ -403,6 +403,44 @@ fn get_characters(event: id) -> Option<String> {
     }
 }
 
+fn get_characters_ignoring_modifiers(event: id) -> Option<String> {
+    unsafe {
+        let characters: id = msg_send![event, charactersIgnoringModifiers];
+        let slice = slice::from_raw_parts(
+            characters.UTF8String() as *const c_uchar,
+            characters.len(),
+        );
+        let string = str::from_utf8_unchecked(slice);
+        Some(string.to_owned())
+    }
+}
+
+// Translate macos scancode into the corresponding windows key code.
+fn to_translated_macos_scancode(event: id) -> c_ushort {
+    // Cmd switches Roman letters for Dvorak-QWERTY layout, so try modified characters first.
+    let characters = get_characters(event);
+    if let Some(chars) = characters {
+        let code = chars.chars().next().map_or(0, |c| windows_key_code_for_char_code(c).unwrap_or(0));
+        if code != 0 {
```

Comment:
This pattern (using `0` as a magic constant) is super unidiomatic, so it would be way nicer to use `Option` throughout this function. You can use [`and_then`](https://doc.rust-lang.org/core/option/enum.Option.html#method.and_then) to chain the result of the first attempt to get a keycode with the result of the second attempt, and then use [`unwrap_or_else`](https://doc.rust-lang.org/core/option/enum.Option.html#method.unwrap_or_else) to default to the scancode at the end (though I'm not sure how relevant this advice will be after changing `windows_key_code_for_char_code` to return `Option<VirtualKeyCode>`).

There's also some repetition in this function that can be simplified out to make things easier to understand (maybe just using an inline function?). That will make it easier to change this into an `Option` chain, too.